### PR TITLE
CI: Test all supported Python versions

### LINKED ISSUE #9: Test all supported Python versions in CI
Add a matrix to the `test` CI job to run on all supported Python versions: https://github.com/commaai/body-jim/blob/2f06d4038c6d440ad2b023c4d2e8848e7d41423c/.github/workflows/tests.yaml#L6

GitHub has some docs here: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#using-multiple-python-versions

### PR DESCRIPTION
**Pull request by incognitojam**
Test 3.8 -> 3.12 (https://devguide.python.org/versions/)

- Depends on https://github.com/commaai/teleoprtc/pull/7
- Closes #9

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ authors = [{ name="Vehicle Researcher", email="user@comma.ai" }]
 description = "Comma body gymnasium API"
 readme = "README.md"
 license = { file="LICENSE" }
-requires-python = ">=3.11"
+requires-python = ">=3.8"
```
Comment (lsrugo) [inline pyproject.toml:12]: Why

**Comment by adeebshihadeh** [comment]
3.8 sounds like a good minimum. If we have to change minor stuff to support it, that's fine.

**Comment by incognitojam** [comment]
We'd have to change back to the old way of writing Optional and Union types

**Comment by adeebshihadeh** [comment]
Either sounds fine. These are relatively small packages.

**Comment by adeebshihadeh** [comment]
Let's also disable fail fast so that all versions are always tested.

**Comment by incognitojam** [comment]
Need to bump teleoprtc version and publish the package

**Comment by incognitojam** [comment]
Should be good now

**Pull request merged**